mycroft/util/parse.py

```diff
@@ -164,6 +169,571 @@ def extractnumber_en(text):
     return val
 
 
+'''authors: spenrod and connorpenrod
+this function takes a sentence and extracts date and time information from it
+returns a 2-element array [date, leftover_string]
+the 'date' is a python datetime object
+the 'leftover_string' contains all keywords not related to the date or time'''
+
+
+def extractdatetime_en(str, currentDate=datetime.now()):
+    if str == "":
+        return None
+    found = False
+    daySpecified = False
+    dayOffset = False
+    monthOffset = 0
+    yearOffset = 0
+    dateNow = currentDate
+    today = dateNow.strftime("%w")
+    currentYear = dateNow.strftime("%Y")
+    fromFlag = False
+    datestr = ""
+    hasYear = False
+    timeQualifier = ""
+
+    str = str.lower().replace('?', '').replace('.', '').replace(',', '')\
+        .replace(' the ', ' ').replace(' a ', ' ').replace(' an ', ' ')
+    words = str.split()
+
+    timeQualifiersList = ['morning', 'afternoon', 'evening']
+    markers = ['at', 'in', 'on', 'by', 'this', 'around', 'for', 'of']
+    days = ['monday', 'tuesday', 'wednesday',
+            'thursday', 'friday', 'saturday', 'sunday']
+    months = ['january', 'february', 'march', 'april', 'may', 'june',
+              'july', 'august', 'september', 'october', 'november', 'december']
+    monthsShort = ['jan', 'feb', 'mar', 'apr', 'may', 'june', 'july', 'aug',
+                   'sept', 'oct', 'nov', 'dec']
+
+    '''strip out 's, plurals, and ordinals (3rd becomes 3)'''
+    for idx, word in enumerate(words):
+        word = word.replace("'s", "")
+
+        ordinals = ["rd", "st", "nd", "th"]
+        if word[0].isdigit():
+            for ord in ordinals:
+                if ord in word:
+                    word = word.replace(ord, "")
+        words[idx] = word
+
+    for idx, word in enumerate(words):
+        if word == "":
+            continue
+
+        wordPrevPrev = words[idx-2] if idx > 1 else ""
+        wordPrev = words[idx-1] if idx > 0 else ""
+        wordNext = words[idx+1] if idx+1 < len(words) else ""
+        wordNextNext = words[idx+2] if idx+2 < len(words) else ""
+        word = word.rstrip('s')
+        start = idx
+        used = 0
+        '''save timequalifier for later'''
```

Comment:
Not a huge deal, but use # for comments instead of strings.  It is ever-so-slightly more efficient, and the floating string is just kinda weird.